glide-core/redis-rs/redis/src/cluster_async/mod.rs

```diff
@@ -3369,6 +3373,123 @@ where
+        .filter_map(|(host, socket_addr)| socket_addr.map(|_| host))
+        .collect();
+
+    if valid_hosts.is_empty() {
+        return Err(RedisError::from((
+            ErrorKind::AllConnectionsUnavailable,
+            "No valid addresses found",
+        )));
+    }
+
+    // Randomly select a subset of valid hosts to query.
+    // The number of hosts selected is determined by `num_of_nodes_to_query`.
+    let selected_hosts: Vec<String> = {
+        let mut rng = rand::rng();
+        valid_hosts
+            .into_iter()
+            .choose_multiple(&mut rng, num_of_nodes_to_query)
+    };
+
+    let cluster_params = Arc::new(inner.cluster_params.read().expect(MUTEX_READ_ERR).clone());
+    let glide_options = Arc::new(inner.glide_connection_options.clone());
+
+    // Create a future for each selected host to establish a connection.
+    let connection_futures = selected_hosts.into_iter().map(|host| {
+        let node_for_addr = {
+            let guard = inner.conn_lock.read().expect(MUTEX_READ_ERR);
+            guard.node_for_address(&host)
+        };
+
+        let host_clone = host.clone();
+        let params_ref = Arc::clone(&cluster_params);
+        let glide_options_ref = Arc::clone(&glide_options);
+
+        async move {
+            match get_or_create_conn::<C>(
+                &host_clone,
+                node_for_addr,
+                &params_ref,
+                RefreshConnectionType::OnlyManagementConnection,
+                (*glide_options_ref).clone(),
+            )
+            .await
+            {
+                Ok(node) => {
+                    let conn = node
+                        .management_connection
+                        .unwrap_or(node.user_connection)
+                        .conn
+                        .await;
+                    Ok((host, conn))
+                }
+                Err(err) => {
+                    tracing::warn!("Failed to create connection for host {}: {:?}", host, err);
+                    Err((host, err))
+                }
+            }
+        }
+    });
+
+    // Resolve all connections and separate successful from failed ones.
+    let results = futures::future::join_all(connection_futures).await;
+
+    let mut successful_connections = Vec::new();
+    let mut failed_connection_addresses = std::collections::HashSet::new();
+
+    for result in results {
+        match result {
+            Ok((host, conn)) => {
+                let future = async move { conn }.boxed().shared();
+                successful_connections.push((host, future));
+            }
+            Err((host, err)) => {
+                if err.is_unrecoverable_error() {
+                    failed_connection_addresses.insert(host);
+                }
+            }
+        }
+    }
+
+    if successful_connections.is_empty() {
+        return Err(RedisError::from((
+            ErrorKind::AllConnectionsUnavailable,
+            "Failed to create any seed-based connections",
```

Comment:
it doesnt mean that all seed nodes were failed. we might do have some other connection of the seed nodes ready. dont use the error, instead think how to handle failures & when would we retry